pkg/assembler/nodes.go

```diff
@@ -16,9 +16,161 @@
 package assembler
 
 // ArtifactNode is a node that represents an artifact
-// TODO: implement GuacNode
-type ArtifactNode struct{}
+type ArtifactNode struct {
+	Name     string
+	Checksum string
+}
+
+func (an ArtifactNode) Type() string {
+	return "Artifact"
+}
+
+func (an ArtifactNode) Properties() map[string]interface{} {
+	properties := make(map[string]interface{})
+	properties["name"] = an.Name
+	properties["checksum"] = an.Checksum
+	return properties
+}
+
+func (an ArtifactNode) Attributes() []string {
+	return []string{"name", "checksum"}
+}
+
+func (an ArtifactNode) IdentifiableAttributes() [][]string {
+	// An artifact can be uniquely identified by checksum
+	return [][]string{{"checksum"}}
+}
 
 // AttestationNode is a node that represents an attestation
-// TODO: implement GuacNode
-type AttestationNode struct{}
+type AttestationNode struct {
+	// TODO(mihaimaruseac): Unsure what fields to store here
+	File     string
+	Checksum string
+}
+
+func (an AttestationNode) Type() string {
+	return "Attestation"
+}
+
+func (an AttestationNode) Properties() map[string]interface{} {
+	properties := make(map[string]interface{})
+	properties["file"] = an.File
+	properties["checksum"] = an.Checksum
+	return properties
+}
+
+func (an AttestationNode) Attributes() []string {
+	return []string{"file", "checksum"}
+}
+
+func (an AttestationNode) IdentifiableAttributes() [][]string {
+	// An attestation can be uniquely identified by filename?
+	return [][]string{{"file"}}
+}
+
+// BuilderNode is a node that represents an attestation
+type BuilderNode struct {
+	BuilderType string
+	BuilderId   string
+}
+
+func (bn BuilderNode) Type() string {
+	return "Builder"
+}
+
+func (bn BuilderNode) Properties() map[string]interface{} {
+	properties := make(map[string]interface{})
+	properties["type"] = bn.BuilderType
+	properties["id"] = bn.BuilderId
+	return properties
+}
+
+func (bn BuilderNode) Attributes() []string {
+	return []string{"type", "id"}
+}
+
+func (bn BuilderNode) IdentifiableAttributes() [][]string {
+	// A builder needs both type and id to be identified
+	return [][]string{{"type", "id"}}
+}
+
+// AttestationEdge is an edge that represents the fact that an
+// `AttestationNode` is an attestation for an `ArtifactNode`.
+type AttestationEdge struct {
```

Comment:
Suggest `AttestationForEdge`, so that it can be read as "`AttestationNode AttestationForEdge ArtifactNode`". This would make it more consistent with `BuiltByEdge`, as in "`ArtifactNode BuiltByEdge BuilderNode`".